src/OpenTelemetry/Batch.cs

```diff
@@ -0,0 +1,129 @@
+﻿// <copyright file="Batch.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry
+{
+    /// <summary>
+    /// Stores a batch of completed <typeparamref name="T"/> objects to be exported.
+    /// </summary>
+    /// <typeparam name="T">The type of object in the <see cref="Batch{T}"/>.</typeparam>
+    public readonly struct Batch<T>
```

Comment:
Is there a way we can mark public APIs as "public for plugin developers only, not for end users"? @cijothomas 
 @eddynaka

Alternative way would be put these in a special namespace such like `OpenTelemetry.ExtensibilityHighVoltageBeCareful`.

Reply:
Some namespace other than `OpenTelemetry` would be good.
like `OpenTelemetry.Extensibility....`?. 
Irrespective of which namespace it is, we still follow the semver guidelines ? (no breaking change unless major version is bumped etc.?

Reply:
We could add a <remarks> tag and be clear what are our intentions there.